### ISSUE TITLE
Preload Trained Model and Continue Training with New Learning Rates

### ISSUE THREAD

**Issue by hmehryar** [hmehryar/ml-jet-param-predictor#22]
**Description:**  
Implement a mechanism to optionally preload a previously trained model (e.g., ViT, Mamba, which are the best models so far) before training, as specified in the XML config. If the model exists at the configured path, load it and continue training for a few extra epochs with staged learning rates (starting at `1e-3` and `1e-4`), while keeping the rest of the training process unchanged.

**Requirements:**
1. **XML Config Update:**
   - Add a `<preload>` section under training configuration:
     ```xml
     <preload enabled="true">
         <model_path>/path/to/best_model.pth</model_path>
         <extra_epochs>2</extra_epochs>
         <learning_rates>1e-3,1e-4</learning_rates>
     </preload>
     ```
2. **Model Preload Logic:**
   - Before training starts, check if `<preload enabled="true">` and if `<model_path>` exists.
   - Load model weights into the architecture defined in the current run.
   - Ensure compatibility for different architectures (strict vs non-strict loading).
3. **Extra Epoch Training:**
   - Train for the specified `<extra_epochs>` using staged learning rates from `<learning_rates>`.
   - Maintain the same dataset, loss, metrics, and logging pipeline as in normal training.
4. **Resume Training Behavior:**
   - If preload is enabled, start from epoch 0 for the extra training, then proceed with the standard training loop.
5. **Logging & Checkpointing:**
   - Clearly log when a pretrained model is loaded, its path, and the learning rates applied.
   - Save checkpoints after each extra epoch.

**Acceptance Criteria:**
- Pretrained model is successfully loaded when config is enabled and file exists.
- Extra epochs are run with specified staged learning rates before main training.
- All logs and checkpoints reflect preloading and staged LR phases.
- Code runs seamlessly whether preload is enabled or disabled.

**Related Files to Modify:**
- `train.ipynb` / training script entry point
- XML parsing module
- Training loop
- Checkpoint loading utilities

**Priority:** High  
**Labels:** `feature`, `training`, `config`, `model-preload`
